### ISSUE TITLE
[Bug] mdx stories with Chinese name are not showing up in side-menu

### ISSUE THREAD

**Issue by youngboy** [storybookjs/builder-vite#389]
### What version of `vite` are you using?

2.9.6

### System info and storybook versions

Environment Info:
  System:
    OS: macOS 12.2
    CPU: (8) x64 Apple M1 Pro
  Binaries:
    Node: 16.10.0 - /usr/local/bin/node
    Yarn: 3.1.1 - /usr/local/bin/yarn
    npm: 7.24.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 101.0.4951.64
    Firefox: 88.0.1
    Safari: 15.3

### Describe the Bug

write stories in mdx format, if the name contains Chinese string (or non-ascii string ?). it will not showing up properly.

## reproduce steps:

clone this repo, add file `examples/vue/stories/abc.stories.mdx` with following content

```mdx
import { Meta, Story } from '@storybook/addon-docs'

import MyButton from './Button.vue';

<Meta title="Example/Utf" component={MyButton} />

export const Template = (args) => ({
  components: { MyButton },
  setup() {
    return { args };
  },
  template: '<my-button v-bind="args" />',
});

<Story
  name="测试"
>
  {Template.bind({})}
</Story>
```

1. run `yarn example:vue` 
2. open http://localhost:6009/
3. story `Example/Utf` not showing as expected
4. modify the file, replace `测试` to `test`
5. `Example/Utf` is showing now

### Participation

- [X] I am willing to submit a pull request for this issue.

after some investigation, I found out the story name occur in https://github.com/storybookjs/builder-vite/blob/main/packages/builder-vite/inject-export-order-plugin.ts#L24 is not equal to https://github.com/storybookjs/storybook/blob/56b5552da077f48bf653b2a115f21d38e95ccf2e/lib/client-api/src/StoryStoreFacade.ts#L177 

<img width="1432" alt="image" src="https://user-images.githubusercontent.com/2573063/169338291-9d096de7-d4e6-42ef-b9cc-a7ceaf9584e3.png">

**Comment by IanVS** [comment]
@youngboy thanks for reporting this. Does it work correctly if you use the webpack builder?

**Comment by youngboy** [comment]
> 

I'm stick with vite, haven't try yet 😂

**Comment by IanVS** [comment]
I wasn't asking you to change to webpack, I meant it as a way to troubleshoot whether it is an issue within storybook itself, or the vite builder (this project).  :-D

**Comment by youngboy** [comment]
I think I found the issue causing this.  https://github.com/brillout/vite-plugin-mdx/blob/master/src/transform.ts#L39

need to add charset: 'utf8' then it will working correctly.

**Comment by IanVS** [comment]
Thanks for the investigation, @youngboy.  We are in the process of moving away from vite-plugin-mdx, see https://github.com/storybookjs/builder-vite/pull/377.  It still needs some kinks worked about before it's ready, but we're working on it.  For now, you may need to stick with the webpack builder if this is a requirement for you.

**Comment by joshwooding** [comment]
Hi @youngboy. This should be fixed in [v0.1.38](https://github.com/storybookjs/builder-vite/releases/tag/v0.1.38)+

Closing this for now but if you still see this please let me know.
